lib/ansible/parsing/dataloader.py

```diff
@@ -171,9 +171,16 @@ def set_basedir(self, basedir):
             self._basedir = to_text(basedir)
 
     def path_dwim(self, given):
-        '''
-        make relative paths work like folks expect.
-        '''
+        """Make relative paths work like folks expect.
+
+        :parama given: The path
+        :type data: str
+
+        :return: The modify path
+        :rtype: str
+        """
+        if not isinstance(given, string_types):
+            raise TypeError('The type of the given argument must be a string')
```

Comment:
Can you explain what led to needing this change?  Where is a non string being passed into this function?

Reply:
Sorry I'm not sure to understand what do you mean, but it's just to be sure it's a string. Sorry for the delay of my answer, I'm going to be more proactive.

Reply:
This PR, per the description is "Fix the unquote function". As such, there is no explanation why a change was made to `path_dwim`

In what case is a non string being passed to the `path_dwim` method?  What code path or behavior leads to this being necessary?

Reply:
If you want I can make this modification on another contribution. But I think it's good to check to type here also and fix the documentation.

Reply:
You still aren't answering my question.

In what case is a non string being passed to the path_dwim method? What code path or behavior leads to this being necessary?

Reply:
The same of the initial fix, a bug with ansible-lint. But I'm not sure to understand why it's important.

Reply:
While fixing bugs is important, we typically do not prioritize bugs that are the fault of other software, or software that misuses the Ansible Python API.  Ansible itself does not suffer from these problems, thus it sounds like the actual bug is really in ansible-lint that is causing issues.

Making these changes, could result in bad behavior of Ansible itself, since we may be handling this differently elsewhere.

Reply:
Sorry you're right, I understand what you mean. So I can move this modification in another contribution or remove it simply, if you want and keep only the unquote function modification.